pulp_rpm/app/downloaders.py

```diff
@@ -0,0 +1,49 @@
+from logging import getLogger
+from aiohttp.client_exceptions import ClientResponseError
+from pulpcore.plugin.download import HttpDownloader
+from urllib.parse import urljoin
+
+log = getLogger(__name__)
+
+
+class RpmDownloader(HttpDownloader):
+    """
+    Custom Downloader that automatically handles authentication token for SLES repositories.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the downloader.
+        """
+        if 'sles_auth_token' in kwargs.keys():
+            self.sles_auth_token = kwargs.pop('sles_auth_token')
```

Comment:
Why do you need to remove `sles_auth_token` from the kwargs?

Reply:
no need to remove but still need to "transfer" to class variable, as another class method using it. In this case I think doesn't matter if `kwargs.pop()` or `kwargs['sles...']` is used for assign it.

Reply:
Fair enough. I try not to change any incoming objects if I don't really need to.